experimental/hal_executable_library_call_hooks/stats.cc

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "experimental/hal_executable_library_call_hooks/stats.h"
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+
+float sum(const std::vector<float> &v) {
```

Comment:
How about using `std::accumulate`?

Reply:
i'm not super hot about fancy c++ features to replace a one-line loop. we don't even care about the performance of that loop.